modules/src/ics02_client/client_def.rs

```diff
@@ -256,7 +266,7 @@ impl From<AnyConsensusState> for Any {
 
 impl ConsensusState for AnyConsensusState {
     fn client_type(&self) -> ClientType {
-        todo!()
+        self.client_type()
```

Comment:
Isn't this a recursive call the the same method?

Reply:
I think it calls the `AnyConsensusState` impl at: https://github.com/informalsystems/ibc-rs/blob/9f661824d02290ea67fdf0fbcfa5c9752e681989/modules/src/ics02_client/client_def.rs#L136
Btw, we have the same pattern for `latest_height()`

Reply:
TIL that's actually not a recursive call: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d4a17a4674079b43f8c460d113a2706f